src/main/java/org/gridsuite/filter/utils/expertfilter/FilterCycleDetector.java

```diff
@@ -36,30 +33,34 @@ private FilterCycleDetector() {
      * @param filterLoader loader used to retrieve referenced filters
      */
     public static void checkNoCycle(AbstractFilter filter, FilterLoader filterLoader) {
-        checkNoCycle(filter, filterLoader, new HashSet<>());
+        checkNoCycle(filter, filterLoader, new ArrayDeque<>());
```

Comment:
addLast and removeLast are part of List interface now, so maybe I would just use List interface with a simple ArrayList